src/Common/tests/Tests/System/SpanTestHelpers.cs

```diff
@@ -26,5 +28,21 @@ public static class SpanTestHelpers
                 Assert.False(true, "Wrong exception thrown: Expected " + typeof(E).GetType() + ": Actual: " + wrongException.GetType());
             }
         }
+
+        public static void AssertThrows<E, T>(Span<T> span, AssertThrowsAction<T> action) where E : Exception
+        {
+            try
+            {
+                action(span);
+                Assert.False(true, "Expected exception: " + typeof(E).GetType());
+            }
+            catch (E)
+            {
+            }
+            catch (Exception wrongException)
+            {
+                Assert.False(true, "Wrong exception thrown: Expected " + typeof(E).GetType() + ": Actual: " + wrongException.GetType());
+            }
+        }
     }
```

Comment:
I think it'd be better to have all of these asserts helpers on AssertExtensions, so that all of our custom assert helpers are in the same place and are immediately available to all projects.